### PR TITLE
fix(ark-metadata): add missing `find_token_ids_without_metadata` function

### PR DESCRIPTION
**Pull request by remiroyc**
## Description

Add a missing function to the ark-metadata storage and rename the struct for Image Cache Options.

## What type of PR is this? (check all applicable)

- [x] 🐛 Bug Fix (`fix:`)

## Added tests?

- [X] 🙅 no, because they aren't needed

## Added to documentation?

- [X] 🙅 no documentation needed

### CODE REVIEW

File: crates/ark-metadata/src/metadata_manager.rs [hunk 1]
```diff
@@ -171,16 +171,27 @@ impl<'a, T: Storage, C: StarknetClient, F: FileManager> MetadataManager<'a, T, C
...
                 })
             }
-            _ => {
-                debug!("Fetching image... {}", url);
-                let response = reqwest::get(url).await?;
+            ImageCacheOption::Save => {
+                let response = self
+                    .request_client
+                    .get(url)
+                    .timeout(Duration::from_secs(5))
```
Comment (glihm) [inline crates/ark-metadata/src/metadata_manager.rs:178]: I would use it as an argument to ensure we can change this later with very few changes.

File: crates/ark-metadata/src/metadata_manager.rs [hunk 2]
```diff
@@ -171,16 +171,27 @@ impl<'a, T: Storage, C: StarknetClient, F: FileManager> MetadataManager<'a, T, C
...
+
+                debug!(
+                    "Image: Content-Type={}, Content-Length={}, File-Ext={}",
+                    content_type, content_length, file_ext
+                );
 
                 self.file_manager
                     .save(&FileInfo {
-                        name: format!("{}.{}", token_id.to_hex(), file_ext),
+                        name: format!("{}.{}", token_id.to_decimal(false), file_ext),
```
Comment (glihm) [inline crates/ark-metadata/src/metadata_manager.rs:194]: Saving in hex would be better for formatting and consistency. Anyway in lambdas APIs we have functions to convert a decimal or unpadded hex into correctly padded hex.

**Review by glihm — APPROVED** [review]
@remiroyc the comments are not mandatory just remarks.
I'll approve to let you move forward, the modification can be done in an other PR.

File: crates/ark-metadata/src/metadata_manager.rs (hunk 2 above)
Comment (remiroyc) [reply crates/ark-metadata/src/metadata_manager.rs:194]: wdyt @kwiss @YohanTz

**Pull request merged**